### PR TITLE
Scavengers: new game mode

### PR DESCRIPTION
**Pull request by sparkychildcharlie**
Add Team Scavs as well as the necessary supporting commands.

This has Meicoo-chan's seal of approval!

kawaiiiii

### CODE REVIEW

File: server/chat-plugins/scavenger-games.ts [hunk 1]
```diff
@@ -25,9 +25,17 @@ interface GameMode {
...
+	teamAnnounce?: GameModeFunction;
+	getPlayerTeam?: GameModeFunction;
+	advanceTeam?: GameModeFunction;
 	[k: string]: any;
 }
 
+type GameModeFunction = (
+	this: ScavengerGameTemplate,
+	...args: any[]
+) => void;
```
Comment (KrisXV) [inline server/chat-plugins/scavenger-games.ts:37]:
```suggestion
type GameModeFunction = (this: ScavengerGameTemplate, ...args: any[]) => void;
```

File: server/chat-plugins/scavengers.ts [hunk 2]
```diff
@@ -663,7 +686,10 @@ export class ScavengerHunt extends Rooms.RoomGame {
 					}</details>` :
 					``)
 			}</td>` +
-			`</tr></table></div>`
+			`</tr></table></div>`;
+
+		player.sendRoom(
+			questionDisplay
 		);
```
Comment (KrisXV) [inline server/chat-plugins/scavengers.ts:693]:
```suggestion
		player.sendRoom(questionDisplay);
```

File: server/chat-plugins/scavengers.ts [hunk 3]
```diff
@@ -652,8 +676,7 @@ export class ScavengerHunt extends Rooms.RoomGame {
 
 		const finalHint = current.number === this.questions.length ? "final " : "";
 
-		player.sendRoom(
-			`|raw|<div class="ladder"><table><tr>` +
+		const questionDisplay = `|raw|<div class="ladder"><table><tr>` +
```
Comment (KrisXV) [inline server/chat-plugins/scavengers.ts:679]: There's a ton of unnecessary new lines in this

Comment (sparkychildcharlie) [reply server/chat-plugins/scavengers.ts:679]: @Zarel wrote this

Comment (KrisXV) [reply server/chat-plugins/scavengers.ts:679]: Yes, but that doesn't mean the new lines are necessary? It's a variable that's separately declared now so the line length shouldn't affect it I'm pretty sure.

Comment (KrisXV) [reply server/chat-plugins/scavengers.ts:679]: The new lines were necessary when it was still wrapped in the function.

Comment (Zarel) [reply server/chat-plugins/scavengers.ts:679]: I definitely prefer the newlines for readability.

Comment (Zarel) [reply server/chat-plugins/scavengers.ts:679]: I don't like the addition of the `questionDisplay` variable, though. What's the point?

**Pull request merged**